src/app/services/article-comparison.service.ts

```diff
@@ -182,4 +182,24 @@ export class ArticleComparisonService {
             .filter((char) => char.match(/[a-zA-Z0-9]/))
             .join("");
     }
+
+    //
+    // Explicit type narrowing because Angular directives _still_ do not
+    // support discriminated unions...
+    //
+    asInsert(edit: Edit): Insert {
```

Comment:
Main thing here is that the Replace type has `from` and `to` as properties rather than `word`...

Discriminated unions are quite powerful, but it's unfortunate that it still isn't supported in the Angular HTML markup... ( +1 to React 😄  )